### ISSUE TITLE
[Feature Request] Turn off truffle compilation dynamically

### ISSUE THREAD

**Issue by TheAxiks** [oracle/graal#9494]
**Feature request**

Please include the following information:

**Is your feature request related to a problem? Please describe.**
We develop a software to control automated vehicles, robots and machines. Inside are multiple scripts (written in JavaScript) to build up our customers needs. Each Script runs with an own engine in multiple threads. The software runs normally without a reboot for multiple weeks. Performance fluctuations can inflict the whole production and traffic a lot and can produce multiple further problems. We have some customers running in performance issues caused by compilations after some days and running in a loop of opt-deopt. The compilation helps a lot to save performance and is necessary. We would like to turn the compilation off for a specific engine dynamically from Java or by time.

TLDR;
After some time (days) the compilation loops in a opt-deopt. This causes performance issues. We would like to turn of the compilation dynamically from Java or by time.

**Describe the solution you'd like.**
We would like to have something like that:

```
Engine engine = Engine.newBuilder().option("engine.CompilationTimeout", "60000").build(); // Turn of compilation after 1 minute for this engine
```

or

```
Engine engine = getEngine();
if (condition == true) engine.turnOffCompilation();
```

**Describe who do you think will benefit the most.**
Everyone. It's a nice option and can maybe fix problems or performance issues.

**Describe alternatives you've considered.**
It can also be possible to make this a global option for every engine.

**Additional context.**
We can have a chat where we show our software and customer projects that will profit from this solution.

**Express whether you'd like to help contributing this feature**
If you'd like to contribute, please read the [contribution guide](https://www.graalvm.org/community/contributors/).

Best regard
Patrick
